for-research/public-likelihoods/public-likelihoods.ipynb

```diff
@@ -0,0 +1,936 @@
+      "            OptimizeConst :               0\n",
+      "            TrackProgress :              30\n",
+      "            BoundaryCheck :               0\n",
+      "Last Rebuild Log Output: \n"
+     ]
+    }
+   ],
+   "source": [
+    "nll = w[\"models/simPdf\"].nll(\"obsData\",nllOpts)\n",
+    "nll.Print() # Prints the PDF, Dataset, NLL Options, and Fit Config"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4873aa37-5437-454b-8a7f-b63ef0ca187b",
+   "metadata": {},
+   "source": [
+    "## 3.4. Evaluating the NLL function\n",
+    "\n",
+    "The NLL function can be evaluated while varying some of the parameters of the model:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "a734911e-31dd-41b9-bc2b-94453e22b624",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "dPhi_ttbar_Z_Bin_001_mu = 0.0  nll= 9511.889887190646\n",
+      "dPhi_ttbar_Z_Bin_001_mu = 0.2222222222222222  nll= 9491.954732711303\n",
+      "dPhi_ttbar_Z_Bin_001_mu = 0.4444444444444444  nll= 9477.590868539133\n",
+      "dPhi_ttbar_Z_Bin_001_mu = 0.6666666666666666  nll= 9467.839443849427\n",
+      "dPhi_ttbar_Z_Bin_001_mu = 0.8888888888888888  nll= 9462.106333762014\n",
+      "dPhi_ttbar_Z_Bin_001_mu = 1.1111111111111112  nll= 9460.019964270667\n",
+      "dPhi_ttbar_Z_Bin_001_mu = 1.3333333333333333  nll= 9461.358980950059\n",
+      "dPhi_ttbar_Z_Bin_001_mu = 1.5555555555555554  nll= 9466.01480265484\n",
+      "dPhi_ttbar_Z_Bin_001_mu = 1.7777777777777777  nll= 9473.974311478687\n",
+      "dPhi_ttbar_Z_Bin_001_mu = 2.0  nll= 9485.316708851196\n",
+      "[#1] INFO:Minimization -- RooNLLVar::evaluatePartition(reg_3l_WZ) first = 0 last = 5 Likelihood offset now set to 405.768\n",
+      "[#1] INFO:Minimization -- RooNLLVar::evaluatePartition(reg_3l_tZq) first = 0 last = 5 Likelihood offset now set to 870.406\n",
+      "[#1] INFO:Minimization -- RooNLLVar::evaluatePartition(reg_3l_ttZ) first = 0 last = 5 Likelihood offset now set to 1302.29\n",
+      "[#1] INFO:Minimization -- RooNLLVar::evaluatePartition(reg_4l_DF) first = 0 last = 5 Likelihood offset now set to 181.195\n",
+      "[#1] INFO:Minimization -- RooNLLVar::evaluatePartition(reg_4l_FF_CR_e_hf) first = 0 last = 1 Likelihood offset now set to 2090.7\n",
+      "[#1] INFO:Minimization -- RooNLLVar::evaluatePartition(reg_4l_FF_CR_e_other) first = 0 last = 6 Likelihood offset now set to 1984.6\n",
+      "[#1] INFO:Minimization -- RooNLLVar::evaluatePartition(reg_4l_FF_CR_mu_hf) first = 0 last = 1 Likelihood offset now set to 1731.69\n",
+      "[#1] INFO:Minimization -- RooNLLVar::evaluatePartition(reg_4l_SF) first = 0 last = 5 Likelihood offset now set to 194.77\n",
+      "[#1] INFO:Minimization -- RooNLLVar::evaluatePartition(reg_4l_ZZ) first = 0 last = 5 Likelihood offset now set to 191.157\n"
+     ]
+    }
+   ],
+   "source": [
+    "import numpy as np\n",
```

Comment:
Do we have to use numpy for this? Can we explain why? That's pretty non-intuitive I think - if it's just for the range, then we could do this in pure python, right? Just as: 
```
for val in [ poi.getMin()+x/10.*(poi.getMax()-poi.getMin()) for x in range(11) ]
```

for example?